DDG4/lcio/LCIOEventReader.cpp

```diff
@@ -146,11 +146,10 @@ LCIOEventReader::readParticles(int event_number,
       vtx->y = p->vsy ;
       vtx->z = p->vsz ;
       vtx->time = p->time ;
-      haveVertex = false ;
+      haveVertex = true ;
```

Comment:
Additionally  the `genStatus == 1` check does not cover all the cases, imagine haveing a K_L